src/main/java/impl/weather/impl/YearTemperatureStatsParserImpl.java

```diff
@@ -0,0 +1,24 @@
+package impl.weather.impl;
+
+import api.weather.YearTemperatureStats;
+import api.weather.YearTemperatureStatsParser;
+
+import java.time.Month;
+import java.util.Collection;
+
+public class YearTemperatureStatsParserImpl implements YearTemperatureStatsParser {
+    @Override
+    public YearTemperatureStats parse(Collection<String> rawData) {
+        var stats = new YearTemperatureStatsImpl();
+//        Could have wrapped this into stream forEach or something like that, but it was way too ugly
+        for (var string : rawData) {
+//          We can try to implement full parser here or at least regex-based scanf analog, but that's gonna be too much
+            var pieces = string.split("\\.| ");
```

Comment:
We have the specific grammar, so we can use split here, because we are sure that this string will be short